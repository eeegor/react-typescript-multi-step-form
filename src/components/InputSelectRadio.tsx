/**
 * Component InputSelectRadio
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { InputGroupRadio } from './InputGroupRadio';
import './InputSelectRadio.scss';

export interface Props {
	id: string;
	name: string;
	label: string;
	value?: string;
	values: string[];
	type?: string;
	info?: string;
	status?: 'success' | 'danger';
	defaultChecked?: string;
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface State {}

export class InputSelectRadio extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);
	}

	render(): JSX.Element {
		const { id, onChange, label, name, defaultChecked, info, status, values } = this.props;
		const cssClass = classnames(
			`input-select input-select-radio`,
			status && `input-select--${status}`
		);

		return (
			<div className={cssClass}>
				<label className="label" htmlFor="salary">
					{label}
				</label>
				{info && <small className="input-select__info">{info}</small>}
				{values.map((item: string, index: number) => (
					<InputGroupRadio
						key={index}
						name={name}
						label={item}
						value={item}
						id={`${id}-${index}`}
						defaultChecked={defaultChecked === item}
						onChange={(radio, event) => onChange && onChange(radio, event)}
					/>
				))}
			</div>
		);
	}
}
