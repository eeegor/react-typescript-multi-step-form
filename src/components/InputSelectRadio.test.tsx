import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { InputSelectRadio } from './InputSelectRadio';
import { mount } from 'enzyme';

describe('InputSelectRadio', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		items: ['one', 'two'],
		onChange: callback,
	};

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<InputSelectRadio {...testData} />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('should adopt a valid status attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputSelectRadio status={status} {...testData} />);
		expect(wrapper.find('.input-select').hasClass('input-select--danger')).toEqual(true);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputSelectRadio info={info} {...testData} />);
		expect(wrapper.find('.input-select__info').text()).toEqual('test');
	});

	it('change event works', () => {
		const wrapper = mount(<InputSelectRadio {...testData} />);
		wrapper
			.find('input[type="radio"]')
			.first()
			.simulate('change', { target: { checked: true } });
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
