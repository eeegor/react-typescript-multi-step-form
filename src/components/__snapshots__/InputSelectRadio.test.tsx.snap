// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelectRadio renders markup correctly 1`] = `
<div
  className="input-select input-select--radio"
>
  <label
    className="label"
    htmlFor="salary"
  >
    Your Name
  </label>
  <div
    className="input-group input-group--radio"
  >
    <input
      className="input-group__input input-radio"
      defaultChecked={false}
      id="name-0"
      name="name"
      onChange={[Function]}
      type="radio"
      value="one"
    />
    <label
      className="input-group__label"
      htmlFor="name-0"
    >
      one
    </label>
  </div>
  <div
    className="input-group input-group--radio"
  >
    <input
      className="input-group__input input-radio"
      defaultChecked={false}
      id="name-1"
      name="name"
      onChange={[Function]}
      type="radio"
      value="two"
    />
    <label
      className="input-group__label"
      htmlFor="name-1"
    >
      two
    </label>
  </div>
</div>
`;
